### PR TITLE
A3 ides 687 state changed

### PR DESCRIPTION
**Pull request by joshymjose**

### CODE REVIEW

File: src/common/marlin_server.cpp [hunk 1]
```diff
@@ -71,12 +71,12 @@ typedef struct _marlin_server_t {
     uint8_t pqueue_tail;                                     // copy of planner.block_buffer_tail
     uint8_t pqueue;                                          // calculated number of records in planner queue
     uint8_t gqueue;                                          // copy of queue.length - number of commands in gcode queue
+    uint8_t server_device_state;                             // device state stored by server
```
Comment (joshymjose) [inline src/common/marlin_server.cpp]: device_state is better ??

File: src/common/marlin_server.cpp [hunk 2]
```diff
@@ -71,12 +71,12 @@ typedef struct _marlin_server_t {
...
     uint32_t command;                                        // actually running command
     uint32_t command_begin;                                  // variable for notification
     uint32_t command_end;                                    // variable for notification
     marlin_mesh_t mesh;                                      // meshbed leveling
     uint64_t mesh_point_notsent[MARLIN_MAX_CLIENTS];         // mesh point mask (points that are not sent)
-    uint8_t device_state_change;                             // marks that change in device's printing state has occured
```
Comment (joshymjose) [inline src/common/marlin_server.cpp:79]: no need of additional mask, there is an array of event masks.
it should be same as how file change mask used

File: src/common/marlin_server.cpp [hunk 3]
```diff
@@ -770,20 +767,18 @@ uint64_t _server_update_vars(uint64_t update) {
     }
 
     if (update & MARLIN_VAR_MSK(MARLIN_VAR_DEVICE_STATE)) {
-        v.ui8 = marlin_server.vars.device_state;
+        v.ui8 = marlin_server.server_device_state;
```
Comment (joshymjose) [inline src/common/marlin_server.cpp]: why not using _device_state_change

File: src/wui/http/http_client.c [hunk 4]
```diff
@@ -435,32 +437,25 @@ err_t data_received_fun(void *arg, struct tcp_pcb *tpcb, struct pbuf *p, err_t e
...
-    snprintf(header, HEADER_MAX_SIZE, "POST %s HTTP/1.0\r\nHost: %s\nPrinter-Token: %s\r\n", uri, host_ip4_str, printer_token);
-    
-    return get_update_str(header);
-}
-
-static const char * events_to_http_str(void * container){
-    
-    char printer_token[API_TOKEN_LEN + 1]; // extra space of end of line
-    char header[HEADER_MAX_SIZE];
+static void create_http_header(char * dest, const char * method, const char * uri, const char * content_type, const char * host_ip){
```
Comment (joshymjose) [inline src/wui/http/http_client.c:447]: investigate if possible to use same as in http server requests, or POST requests

File: src/wui/wui.c [hunk 5]
```diff
@@ -18,7 +18,7 @@
 #define MAX_WUI_REQUEST_LEN       100
 #define MAX_MARLIN_REQUEST_LEN    100
 #define WUI_FLG_PEND_REQ          0x0001
-#define BUDDY_DISABLE_HTTP_CLIENT //disabling http client for next public release
+//#define BUDDY_DISABLE_HTTP_CLIENT //disabling http client for next public release
```
Comment (joshymjose) [inline src/wui/wui.c]: remove this

File: src/common/marlin_server.cpp (hunk 1 above)
Comment (michalrudolf) [reply src/common/marlin_server.cpp]: Its just server's variable device_state, named this way to set it apart from marlin_vars_t's device_state

File: src/wui/wui.c (hunk 5 above)
Comment (michalrudolf) [reply src/wui/wui.c]: commit 0cb14560a...

File: src/common/marlin_server.cpp (hunk 3 above)
Comment (michalrudolf) [reply src/common/marlin_server.cpp]: commit 0cb14560a...

File: src/common/marlin_server.cpp (hunk 2 above)
Comment (michalrudolf) [reply src/common/marlin_server.cpp:79]: commit 0cb14560a...

**Pull request closed without merging**